### PR TITLE
#165273568 make descriptive user registration error messages

### PR DESCRIPTION
**Pull request by kalsmic**
#### What does this PR do?
 - Add descriptive user registration messages.

#### Description of tasks to be completed?
- Adds descriptive user registration messages
- Adds tests for user registration password validation

#### How should this manually test?
- Navigate to the user registration endpoint, `/api/users/`
- Test for the different responses the request with the following inputs.

1.Input password with less than 8 characters
2. Input password with non-alphanumeric characters
3. Input password without a lower case character
4. Input password without an upper case character
5.Register user with duplicate email
6. Register user with duplicate username

[#165273568](https://www.pivotaltracker.com/story/show/165273568)

### CODE REVIEW

File: .env_sample [hunk 1]
```diff
@@ -1,5 +1,5 @@
-export DB_NAME='dojo'
-export DB_USER='postgres'
-export DB_PASS='postgress'
-export DB_HOST='localhost'
-export DB_PORT='5432'
\ No newline at end of file
+export DB_NAME=''
```
Comment (JamesMudidi) [inline .env_sample:1]: Its good that you put this back

**Review by JamesMudidi — APPROVED** [review]
I see great work here

File: authors/apps/authentication/tests/test_user_registration.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+
+        response = self.client.post(
+            "/api/users/",
+            content_type="application/json",
+            data=json.dumps(self.user),
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(
+            response.data["errors"]["password"][0],
+            "Password should be alpha Numerics",
```
Comment (bisonlou) [inline authors/apps/authentication/tests/test_user_registration.py]: spelling error: change alpha Numerics to alpha-numeric

Comment (kalsmic) [reply authors/apps/authentication/tests/test_user_registration.py]: Thanks, @bisonlou for the highlight, let me correct it

**Review by bisonlou — CHANGES_REQUESTED** [review]
I have found that I can register with my password similar to the username. This is not proper.
image

**Review by llwasampijja — APPROVED** [review]
I have tested the different scenarios for the registration, and the messages are descriptive enough.
Good work @kalsmic

File: .env_sample (hunk 1 above)
Comment (kalsmic) [reply .env_sample:1]: Thanks, @JamesMudidi, but not every user who wants to try out the app on will have the same database name on their system.

**Review by nadralia — CHANGES_REQUESTED** [review]
Nice Work @kalsmic , So i tested the feature and i discovered that we can create a user that has username with spaces.
Screenshot 2019-04-25 at 10 18 26

**Comment by kalsmic** [comment]
> Nice Work @kalsmic , So i tested the feature and i discovered that we can create a user that has username with spaces.
> Screenshot 2019-04-25 at 10 18 26

Thanks @nadralia, I have worked on this issue

**Comment by kalsmic** [comment]
> I have found that I can register with my password similar to the username. This is not proper.
> image
@bisonlou, I am going to add this to the technical debt so that we can proceed with the other tasks as we figure out how to enforce it.

**Review by nadralia — APPROVED** [review]

**Review by bisonlou — APPROVED** [review]

**Pull request merged**